dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java

```diff
@@ -708,7 +708,10 @@ public void close() throws IOException
             }
 
             this.dos.close();
-            this.dis.close();
+            if (this.dis != null)
```

Comment:
Are the output stream and sock guaranteed to be non-null here?

Reply:
I believe so, as the output stream is created by us. The input stream can be null if there are no records on the node. I'm fine waiting until @jpmcmu  is back to verify this.